commands/update-check.go

```diff
@@ -0,0 +1,121 @@
+package commands
+
+import (
+	"time"
+	"strings"
+	"strconv"
+	"context"
+
+	"github.com/google/go-github/github"
+	"github.com/activecm/rita/resources"
+	"github.com/activecm/rita/config"
+	log "github.com/sirupsen/logrus"
+)
+
+// UpdateCheck Performs a check for the new version of RITA against the git repository and
+//returns a string indicating the new version if available
+func UpdateCheck( configFile string) string {
+	res := resources.InitResources( configFile )
+	deltaPtr := res.Config.S.UserConfig.UpdateCheckFrequency
+	var version string
+	var delta int
+
+	if deltaPtr == nil {
+		delta = 14
+	} else {
+		delta = *deltaPtr
+	}
+
+	if delta <= 0 {
+		return ""
+	}
+
+	//Check Logs for Versioning
+	m := res.MetaDB
+	timestamp, version := m.LastCheck()
+
+	days := time.Now().Sub( timestamp ).Hours()/24
+
+	if days > float64(delta) {
+		version, err := getRemoteVersion()
+
+		if err != nil {
+			return ""
+		}
+
+		//Log checked version.
+		res.Log.WithFields(log.Fields{
+			"Message":  "Checking versions...",
+			"LastUpdateCheck": time.Now(),
+			"NewestVersion": version,
+		}).Info("Checking for new version")
+	}
+
+	index, err := versionCmp( version, config.Version )
+	if err == nil  && index != -1 {
+		return informUser( version, index )
+	}
+
+	return ""
+}
+
+// Checks if s1 is newer than s2
+// Returns the first index where s1 is greater than s2
+// Returns -1 if s2 is newer than s1
+func versionCmp( s1 string, s2 string) (int, error) {
+	arr1 := parseVersion( s1 )
```

Comment:
I'd prefer to use the semver package used elsewhere throughout RITA to keep things consistent. 

See: https://godoc.org/github.com/blang/semver  https://github.com/activecm/rita/blob/master/commands/analyze.go#L74

We probably want to use ParseTolerant and the GT comparison method.